api-tests/src/steps/ipv-steps.ts

```diff
@@ -188,11 +211,23 @@ When(
         this.lastJourneyEngineResponse.client.redirectUrl,
       ),
     );
-    this.identity = await externalClient.getIdentity(tokenResponse);
+
+    if (result === "identity") {
+      this.identity = await externalClient.getIdentity(tokenResponse);
+    }
+
+    if (result === "MFA reset result") {
+      this.mfaResetResult =
+        await externalClient.getMfaResetResult(tokenResponse);
+    }
   },
 );
 
 Then("I get a {string} identity", function (this: World, vot: string): void {
+  if (!this.identity) {
+    throw new Error("No identity found.");
+  }
+
   assert.equal(this.identity.vot, vot);
```

Comment:
(The lazy option would be to change these to `assert.equal(this.identity?.vot, vot)` etc. - but your way gives a slightly clearer error message)